contracts/base/Settings.sol

```diff
@@ -351,6 +359,66 @@ contract Settings is Pausable, SettingsInterface {
         );
     }
 
+    /**
+        @notice Add a new Node Component with its version.
+        @param componentName name of the component to be added.
+        @param minVersion minimum component version supported.
+     */
+    function addNewComponent(bytes32 componentName, uint256 minVersion)
+        external
+        onlyPauser()
+        whenNotPaused()
+    {
+        require(minVersion > 0, "INVALID_COMPONENT_MINIMUM_VERSION");
+        require(componentName[0] != 0, "MISSING_COMPONENT_NAME");
+        require(minimumNodeRequirements[componentName] == 0, "COMPONENT_ALREADY_EXISTS");
+        minimumNodeRequirements[componentName] = minVersion;
+        emit NodeComponentAdded(msg.sender, componentName, minVersion);
+    }
+
+    /**
+        @notice Remove a Node Component from the list.
+        @param componentName name of the component to be removed.
+     */
+    function removeComponent(bytes32 componentName)
+        external
+        onlyPauser()
+        whenNotPaused()
+    {
+        require(componentName[0] != 0, "MISSING_COMPONENT_NAME");
+        require(minimumNodeRequirements[componentName] > 0, "COMPONENT_NOT_FOUND");
+        uint256 previousVersion = minimumNodeRequirements[componentName];
+        delete minimumNodeRequirements[componentName];
+        emit NodeComponentRemoved(msg.sender, componentName, previousVersion);
+    }
+
+    /**
+        @notice Set a new version for a Node Component.
```

Comment:
Should we have a getter (view) function to get the current version for a given component (and in the interface too)?

Reply:
Getter created.